### PR TITLE
WIP: App run as workflow

### PR DESCRIPTION
**Pull request by leejanee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: apis/core.oam.dev/common/types.go [hunk 1]
```diff
@@ -328,12 +339,31 @@ type ApplicationComponent struct {
...
+	Inputs  StepInputs  `json:"inputs,omitempty"`
+	Outputs StepOutputs `json:"outputs,omitempty"`
+
 	// +kubebuilder:pruning:PreserveUnknownFields
 	// scopes in ApplicationComponent defines the component-level scopes
 	// the format is <scope-type:scope-instance-name> pairs, the key represents type of `ScopeDefinition` while the value represent the name of scope instance.
 	Scopes map[string]string `json:"scopes,omitempty"`
 }
 
+type inputItem struct {
```
Comment (hongchaodeng) [inline apis/core.oam.dev/common/types.go:351]: Need to export this -- InputItem

File: apis/core.oam.dev/common/types.go [hunk 2]
```diff
@@ -328,12 +339,31 @@ type ApplicationComponent struct {
...
+	ParameterKey string `json:"parameterKey"`
+	From         string `json:"from"`
+}
+
+// StepInputs defines variable input of WorkflowStep
+type StepInputs []inputItem
+
+type outputItem struct {
```
Comment (hongchaodeng) [inline apis/core.oam.dev/common/types.go:359]: Need to export this -- OutputItem

File: apis/core.oam.dev/common/types.go (hunk 1 above)
Comment (hongchaodeng) [reply apis/core.oam.dev/common/types.go:351]: Because StepInputs is public type

File: pkg/workflow/providers/oam/parser.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	wfTypes "github.com/oam-dev/kubevela/pkg/workflow/types"
+)
+
+const (
+	// ProviderName is provider name for install.
+	ProviderName = "oam"
+)
+
+// Parser parse types.ComponentManifest from application's component.
+type Parser func(ctx context.Context, comp common.ApplicationComponent) (*types.ComponentManifest, error)
```
Comment (hongchaodeng) [inline pkg/workflow/providers/oam/parser.go:39]: Maybe rename to ComponentParser?

**Comment by wonderflow** [comment]
I'm working on it and this is not ready for review now

**Pull request closed without merging**